utils/record/record.go

```diff
@@ -167,25 +206,23 @@ func NewAPIFactorySettings() api.Settings {
 
 // Send notifications for triggered event if user is subscribed
 func (e *EventRecorderAdapter) sendNotifications(object runtime.Object, opts EventOptions) error {
+	notificationsAPI, err := e.apiFactory.GetAPI()
+	if err != nil {
+		return err
```

Comment:
One minor issue here. We are going to pollute controller logs with error messages if notifications are not configured at all.  The error message should be printed only if someone explicitly added subscribe annotation and notifications are not configured. You can use the following snippet to avoid printing unnecessary error message:

```golang
import "k8s.io/apimachinery/pkg/api/errors"
```
```golang
// don't return error if notifications are not configured and rollout has no subscribers
subsFromAnnotations := subscriptions.Annotations(object.(metav1.Object).GetAnnotations())
if errors.IsNotFound(err) && len(subsFromAnnotations.GetDestinations(nil, map[string][]string{})) == 0 {
	return nil
}
return err
```

Reply:
sure sounds good.